client/my-sites/stats/activity-log/index.jsx

```diff
@@ -193,14 +218,16 @@ class ActivityLog extends Component {
 			startDate,
 		} = this.props;
 
-		const startOfMonth = moment( startDate ).startOf( 'month' ),
-			startOfMonthMs = startOfMonth.valueOf(),
-			endOfMonthMs = moment( startDate ).endOf( 'month' ).valueOf();
-		const filteredLogs = filter( logs, obj => startOfMonthMs <= obj.ts_site && obj.ts_site <= endOfMonthMs );
-		const logsGroupedByDate = map(
+		const siteOffset = this.getSiteOffsetFunc();
+
+		const startOfMonthMs = siteOffset( moment( startDate ).startOf( 'month' ) ).valueOf();
+		const endOfMonthMs = siteOffset( moment( startDate ).endOf( 'month' ) ).valueOf();
```

Comment:
You make a crucial point though, that all the times need to be in the same format (either utc or site) when making comparisons. We can't leave anything in browser time.